File: voiceengine/src/main/java/com/backdoor/engine/lang/WorkerFactory.java
package com.backdoor.engine.lang;

import com.backdoor.engine.misc.Locale;

public class WorkerFactory {

    public static WorkerInterface getWorker(String locale) {
        System.out.println("getWorker: " + locale);
        if (locale.matches(Locale.EN)) {
            return new EnWorker();
        } else if (locale.matches(Locale.UK)) {
            return new UkWorker();
        } else if (locale.matches(Locale.RU)) {
            return new RuWorker();
        } else if (locale.matches(Locale.DE)) {
            return new DeWorker();
        } else if (locale.matches(Locale.ES)) {
            return new EsWorker();
        } else if (locale.matches(Locale.PT)) {
            return new PtWorker();
        } else return new EnWorker();
    }
}

File: voice-engine-ktx/src/main/java/com/backdoor/engine/lang/WorkerFactory.kt
package com.backdoor.engine.lang

import com.backdoor.engine.misc.ContactsInterface
import com.backdoor.engine.misc.Locale
import org.threeten.bp.ZoneId

internal object WorkerFactory {

  fun getWorker(
    locale: String,
    zoneId: ZoneId,
    contactsInterface: ContactsInterface?
  ): WorkerInterface {
    return when (locale) {
      Locale.EN -> EnWorker(zoneId, contactsInterface)
      Locale.UK -> UkWorker(zoneId, contactsInterface)
      Locale.ES -> EsWorker(zoneId, contactsInterface)
      Locale.PT -> PtWorker(zoneId, contactsInterface)
      Locale.PL -> PlWorker(zoneId, contactsInterface)
      Locale.IT -> ItWorker(zoneId, contactsInterface)
      else -> EnWorker(zoneId, contactsInterface)
    }.also {
      println("getWorker: $locale, $it")
    }
  }
}
